android-lib/src/androidTest/java/com/kinvey/androidTest/store/DataStoreTest.java

```diff
@@ -2449,4 +2449,46 @@ public void tearDown() {
         }
     }
 
+    @Test
+    public void testDeltaCache() throws InterruptedException, IOException {
```

Comment:
I don't understand how this is testing delta set caching. I think you should add more thorough tests, ideally with mocked network responses. For example 

- Pull a collection with `n` items into a fresh datastore. Then pull it again. The delta set should be of size zero, and cache should have the `n` items in it.
- Pull a collection with `n` items into a fresh datastore. Add or update an item on the backend, then pull again. The delta should be of size 1, and the new cache should reflect the new state of the backend.
... and so on.

iOS has a set of tests along such lines - https://github.com/Kinvey/swift-sdk/blob/f8302e75ffd11435b51625110dff746c38585a99/Kinvey/KinveyTests/DeltaSetCacheTestCase.swift#L634

Reply:
Before fix the `store.syncBlocking(null)` line had an exception, and here it's checking that after the fix SDK doesn't have this exception.
This is the issue described in the ticket.
There's a ticket for Delta Cache tests' improvements: `MLIBZ-2279 Add tests for CRUD operations with delta set caching enabled`. I've added tests for checking that CRUD operation works if Delta Cache is enabled. https://github.com/Kinvey/java-library/pull/128 . 
About the cases with checking the size of items which were got from the server if Delta Cache is enabled: SDK merges these items with items from cache inside DeltaPull (or DeltaGet) requests, these requests have the private method with the additional request for this. 
I can try to add test to this method, I think it will be good to do in this branch https://github.com/Kinvey/java-library/pull/128. 
But I'm not sure that I'll finish it before Friday's release.

Reply:
Makes sense. I'll review your tests in #128. I understand the tricky part with the private methods - it's not absolutely necessary, but we should add it at some point.
If #128 looks good, we can merge these two open PRs and release.